io/include/traccc/io/read_particles.hpp

```diff
@@ -57,13 +57,15 @@ void read_particles(particle_collection_types::host &particles,
 /// @param[in]  event     The event ID to read in the particles for
 /// @param[in]  directory The directory holding the particle data files
 /// @param[in]  format    The format of the particle data files (to read)
-/// @param[in]  dd        Detector description to re-map particle data with
+/// @param[in]  use_acts_geom_source  Use acts geometry source
+/// @param[in]  detector  detray detector
 /// @param[in]  filename_postfix Postfix for the particle file name(s)
 ///
 void read_particles(particle_container_types::host &particles,
                     std::size_t event, std::string_view directory,
+                    bool use_acts_geom_source = true,
+                    const traccc::default_detector::host *detector = nullptr,
                     data_format format = data_format::csv,
-                    const silicon_detector_description::host *dd = nullptr,
                     std::string_view filename_postfix = "-particles_initial");
```

Comment:
In all of these cases you should just replace the `silicon_detector_description` pointer with a `default_detector` one in place. This would reduce the number of places that you need to update. (All the ones that didn't touch the `dd` pointer before.)

Reply:
This is subjective thing but let's just go this way because I wanted to put detector at the front. It's already quite a work to change this back. Anyway the most of lines should also be changed to set  `use_acts_geom_source` to `false`

Reply:
You misunderstood me. I'm not saying that the command line option should not be there. I'm saying that there is no boolean argument needed in this function.

When calling this function, one would rather do something like:

```c++
traccc::io::read_particles(..., (input_opts.use_acts_geom_source ? &detector : nullptr), ...);
```

Then, in cases where we fundamentally don't have access to a `detray::detector` object, we can just write `nullptr`, and be done with it. Without having to write both `false` and `nullptr`.

My point remains that **in the I/O functions** the two arguments that you introduced are fully coupled. The boolean argument serves no function.

Reply:
Please check my above comment: https://github.com/acts-project/traccc/pull/692#discussion_r1801163133

The boolean argument is required for a good reason.

Reply:
> we can just write nullptr, and be done with it. Without having to write both false and nullptr.

For detray simulation, the arguments are set to `false` and `detector`.  Just for clarification:

1. `false or true`, `nullptr` -> tml csv (which is going to be deprecated at some point)
2. `false`, `detector` -> detray simulation
3. `true`, `detector` -> acts simulation

So what is going to be removed actually is the detector argument. `use_acts_geom_source` should sit there always

Reply:
As discussed in person, Attila's suggestion is taken